release/release.go

```diff
@@ -151,6 +151,62 @@ func CheckUpstreamRelease(ctx context.Context, client *github.Client, org, repo
 	return releases, nil
 }
 
+func KubernetesGoVersion(ctx context.Context, client *github.Client) (map[string]string, error) {
+	const (
+		defaultPerPage int    = 100
+		limitMax       int    = 20
+		owner          string = "kubernetes"
+		repo           string = "kubernetes"
+	)
+	var latestVersion, stableVersion string
+	goVersions := make(map[string]int)
+	maxOccurrences := 0
+	limitCount := 1
+
+	branches, _, err := client.Repositories.ListBranches(ctx, owner, repo, &github.BranchListOptions{
+		ListOptions: github.ListOptions{PerPage: defaultPerPage},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	for i := 0; i < len(branches)/2; i++ {
+		branches[i], branches[len(branches)-1-i] = branches[len(branches)-1-i], branches[i]
+	}
+
+	for _, branch := range branches {
+		if limitCount == limitMax {
+			break
+		}
+		file, _, _, err := client.Repositories.GetContents(ctx, owner, repo, ".go-version", &github.RepositoryContentGetOptions{
+			Ref: *branch.Name,
+		})
+		limitCount++
```

Comment:
Always check the error first before doing anything else.

Reply:
Done